konrad/humidity/relative_humidity.py

```diff
@@ -264,159 +270,206 @@ def __call__(self, atmosphere, **kwargs):
+    def __init__(
+        self,
+        top_peak_T=None,
+        top_peak_rh=0.75,
+        freezing_pt_rh=0.4,
+        bl_top_p=940e2,
+        bl_top_rh=0.85,
+        surface_rh=0.75,
+    ):
         """
-        Defines a C-shaped polynomial model. 
-        The RH increases linearly in the boundary layer and decreases linearly above the upper-tropospheric peak. 
-        The point above the upper-tropospheric peak where the RH is half the peak is coupled to the cold-point.
-        Between the two peaks, a quadratic function is defined by the point of the two peaks and one in the mid-troposphere.
-        Default values from RCEMIP large experiment statistics.
+        Defines a C-shaped polynomial model, that depends on T in the upper troposphere.
+        The RH increases linearly in the boundary layer from the surface.
+        Between the top of the boundary layer and the freezing level (T=273.15K), the rh is a quadratic function of p,
+        defined by its values at these to points, and a zero derivative at the freezing level.
+        Above the freezing level, the rh is a quadratic function of T, defined by its values at the freezing level and
+        at a chose upper-tropospheric T-value or at the cold point (see `top_peak_T` argument), and a zero derivative
+        at the freezing level.
 
         Parameters:
-            top_slope (float): slope of the linear function above the upper-tropospheric peak.
+            top_peak_T (float): Temperature of the upper tropospheric peak. If None, coupled to the cold-point.
             top_peak_rh (float in [0;1]): value of relative humidity at the upper-tropospheric peak.
-            mid_p (float): Pressure of the mid-tropospheric point.
-            mid_rh (float in [0;1]): value of relative humidity at the mid-tropospheric point.
-            low_peak_p (float): Pressure of the low-tropospheric peak.
-            low_peak_rh (float in [0;1]): value of the relative humidity at the low-tropospheric peak. 
-            bl_slope (float): slope of the relative humidity in the boudary layer.
+            freezing_pt_rh (float in [0;1]): value of relative humidity at the freezing point.
+            bl_top_p (float): Pressure of the top of the boundary layer (bl) where the humidity peak is.
+            bl_top_rh (float in [0;1]): value of the relative humidity at the top of the boundary layer.
+            surface_rh (float in [0;1]): value of the relative humidity at the surface.
         """
 
-        ## Convert percent to dimensionless 
-        if top_peak_rh > 1 : top_peak_rh /= 100;
-        if mid_rh > 1 : mid_rh /= 100;
-        if low_peak_rh > 1 : low_peak_rh /= 100;
+        ## Convert percent to dimensionless
+        if top_peak_rh > 1:
+            top_peak_rh /= 100
+        if freezing_pt_rh > 1:
+            freezing_pt_rh /= 100
+        if bl_top_rh > 1:
+            bl_top_rh /= 100
+        if surface_rh > 1:
+            surface_rh /= 100
```

Comment:
Personally, I do not like these "we are silently changing your errors" approaches. Either throw an `ValueError` if any of those is `>1` or just use the values _sh*t in, sh*t out_ ;)

Reply:
I printed a warning message instead, that you might want to make a real python warning.